tests/components/utility_meter/test_sensor.py

```diff
@@ -159,6 +160,67 @@ async def test_state(hass):
     assert state.state == "0.123"
 
 
+async def test_device_class(hass):
+    """Test utility device_class."""
+    config = {
+        "utility_meter": {
+            "energy_meter": {
+                "source": "sensor.energy",
+            },
+            "gas_meter": {
+                "source": "sensor.gas",
+            },
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+    assert await async_setup_component(hass, SENSOR_DOMAIN, config)
+    await hass.async_block_till_done()
+
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    entity_id_energy = config[DOMAIN]["energy_meter"]["source"]
+    hass.states.async_set(
+        entity_id_energy, 2, {ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR}
+    )
+    entity_id_gas = config[DOMAIN]["gas_meter"]["source"]
+    hass.states.async_set(
+        entity_id_gas, 2, {ATTR_UNIT_OF_MEASUREMENT: VOLUME_CUBIC_METERS}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.energy_meter")
+    assert state is not None
+    assert state.state == "0"
+    assert state.attributes.get("device_class") is None
+    assert state.attributes.get("state_class") == "measurement"
```

Comment:
This is needed because there is a race in `UtilityMeterSensor.async_added_to_hass`; I think we should poll the state of the source sensor before subscribing to state change events similar to what is done here:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/switch/light.py#L119-L133

Reply:
Should we fix this in a separate PR?

Reply:
Makes sense to fix it in a separate PR I think.